src/container/RelayRenderer.js

```diff
@@ -116,19 +115,48 @@ const {PropTypes} = React;
  */
 class RelayRenderer extends React.Component {
   mounted: boolean;
+  pendingRequest: ?Abortable;
   props: RelayRendererProps;
   state: RelayRendererState;
 
   constructor(props: RelayRendererProps, context: any) {
     super(props, context);
     this.mounted = true;
-    this.state = this._runQueries(this.props);
+    this.pendingRequest = null;
+    this.state = this._buildState(null, null, null, null);
+  }
+
+  /**
+   * @private
+   */
+  _buildState(
+    activeComponent: ?RelayContainer,
+    activeQueryConfig: ?RelayQueryConfigSpec,
+    readyState: ?ReadyState,
+    props: ?Object
+  ): RelayRendererState {
+    return {
+      activeComponent,
+      activeQueryConfig,
+      readyState: readyState && {...readyState, mounted: true},
+      renderArgs: {
+        done: !!readyState && readyState.done,
+        error: readyState && readyState.error,
+        props,
+        retry: () => this._retry(),
+        stale: !!readyState && readyState.stale,
+      },
+    };
   }
 
   getChildContext(): Object {
     return {route: this.props.queryConfig};
   }
 
+  componentDidMount(): void {
+    this._runQueries(this.props);
```

Comment:
I assume this does not `this.setState(...)` because it is assumed to be identical to the call in `constructor`. Instead of making `this._runQueries` return the new state, how about completely separating the two calls?